docs/cudf/source/user_guide/pandas-comparison.md

```diff
@@ -0,0 +1,155 @@
+# Comparison of cuDF and Pandas
+
+cuDF is a DataFrame library that closely matches the Pandas API, but
+leverages NVIDIA GPUs for performing computations for speed.  However,
+there are some differences between cuDF and Pandas, both in terms of API
+and behavior. This page documents the similarities and differences
+between cuDF and Pandas.
+
+## Supported operations
+
+cuDF supports many of the same data structures and operations as
+Pandas.  This includes `Series`, `DataFrame`, `Index` and
+operations on them such as unary and binary operations, indexing,
+filtering, concatenating, joining, groupby and window operations -
+among many others.
+
+The best way to check if we support a particular Pandas API is to search
+our [API docs](/api_docs/index).
+
+## Data types
+
+cuDF supports many of the commonly-used data types in Pandas,
+including numeric, datetime, timestamp, string, and categorical data
+types.  In addition, we support special data types for decimal, list
+and "struct" values.  See the section on [Data Types](data-types) for
+details.
+
+Note that we do not support custom data types like Pandas'
+`ExtensionDtype`.
+
+## Null (or "missing") values
+
+Unlike Pandas, *all* data types in cuDF are nullable,
+meaning they can contain missing values (represented by `cudf.NA`).
+
+```{code} python
+>>> s = cudf.Series([1, 2, cudf.NA])
+>>> s
+>>> s
+0       1
+1       2
+2    <NA>
+dtype: int64
+```
+
+Nulls are not coerced to `nan` in any situation;
+compare the behaviour of cuDF with Pandas below:
+
+```{code} python
+>>> s = cudf.Series([1, 2, cudf.NA], dtype="category")
+>>> s
+0       1
+1       2
+2    <NA>
+dtype: category
+Categories (2, int64): [1, 2]
+
+>>> s = pd.Series([1, 2, pd.NA], dtype="category")
+>>> s
+0      1
+1      2
+2    NaN
+dtype: category
+Categories (2, int64): [1, 2]
+```
+
+See the docs on [missing data](missing-data) for
+details.
+
+## Iteration
+
+Iterating over a cuDF `Series`, `DataFrame` or `Index` is not
+supported. This is because iterating over data that resides on the GPU
+will yield *extremely* poor performance, as GPUs are optimized for
+highly parallel operations rather than sequential operations.
+
+In the vast majority of cases, it is possible to avoid iteration and
+use an existing function or method to accomplish the same task. If you
+absolutely must iterate, copy the data from GPU to CPU by using
+`.to_arrow()` or `.to_pandas()`, then copy the result back to GPU
+using `.from_arrow()` or `.from_pandas()`.
+
+## Result ordering
```

Comment:
Maybe something like:

> cuDF leverages GPUs to execute operations in parallel. 
> This means the order of operations is not always deterministic. 
> This impacts the determinism of floating-point operations because floating-point arithmetic is non-associative (`a + b != b + a` ).
> For example, computing a `groupby("float column").sum()` is not guaranteed to produce bitwise identical results to Pandas nor produce bitwise identical results from run to run. 
> In general, relying on bitwise equality of floating-point results is not recommended and methods that account for the representations intrinsic imprecision should be used instead.
> For more information see [here](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/).

Reply:
CC @harrism since he's my floating-point guru :)

Reply:
> non-associative (a + b != b + a ).

This example demonstrates non-commutativity. From [Wikipedia](https://en.wikipedia.org/wiki/Floating-point_arithmetic#Accuracy_problems):

> While floating-point addition and multiplication are both [commutative](https://en.wikipedia.org/wiki/Commutative) (a + b = b + a and a × b = b × a), they are not necessarily [associative](https://en.wikipedia.org/wiki/Associative_property). That is, (a + b) + c is not necessarily equal to a + (b + c).

The example of non-associativity at the end is what I would use.

Reply:
lol, I don't think I will ever not mix up "commutative" and "associative".

Reply:
FIx'd